### PR TITLE
SCUMM: Add Hebrew support for MM

### PR DESCRIPTION
**Pull request by ShaharAriel**
- Add MM to existing fixes of RTL support
- Add zak and MM Hebrew preposition

### CODE REVIEW

File: engines/scumm/string.cpp [hunk 1]
```diff
@@ -785,8 +785,8 @@ void ScummEngine::CHARSET_1() {
 		_nextLeft -= _charset->getStringWidth(0, _charsetBuffer + _charsetBufPos) / 2;
 		if (_nextLeft < 0)
 			_nextLeft = _game.version >= 6 ? _string[0].xpos : 0;
-	} else if (_game.version >= 4 && _game.version < 7 && _game.heversion == 0 && _language == Common::HE_ISR) {
-		if (_game.id == GID_MONKEY && _charset->getCurID() == 4) {
+	} else if ((_game.id == GID_MANIAC || (_game.version >= 4 && _game.version < 7)) && _game.heversion == 0 && _language == Common::HE_ISR) {
+		if (_game.id == GID_MANIAC ||(_game.id == GID_MONKEY && _charset->getCurID() == 4)) {
```
Comment (bluegr) [inline engines/scumm/string.cpp:789]:
```suggestion
		if (_game.id == GID_MANIAC || (_game.id == GID_MONKEY && _charset->getCurID() == 4)) {
```

File: engines/scumm/string.cpp [hunk 2]
```diff
@@ -538,8 +538,8 @@ bool ScummEngine::newLine() {
 			// (FT/German, if you look at the sign on the container at game start). After counterchecking
 			// the original code it seems that setting _nextLeft to 0 is the right thing to do here.
 			_nextLeft = /*_game.version >= 6 ? _string[0].xpos :*/ 0;
-	} else if (_game.version >= 4 && _game.version < 7 && _game.heversion == 0 && _language == Common::HE_ISR) {
-		if (_game.id == GID_MONKEY && _charset->getCurID() == 4) {
+	} else if ((_game.id == GID_MANIAC || (_game.version >= 4 && _game.version < 7)) && _game.heversion == 0 && _language == Common::HE_ISR) {
```
[the remark this answers is not in the document]
Comment (orgads) [reply engines/scumm/string.cpp:541]: That's fine. See https://github.com/orgads/scummvm/commit/0cc69edeee7b5cb1cd929af13cabe2b70d39af89#diff-9824334a2501e678c610fb701996a6396dd92fa2a2dcae531d4f64417556b812R958 for a similar example.

[the remark this answers is not in the document]
Comment (orgads) [reply engines/scumm/string.cpp:541]: PR pending: #3882

[the remark this answers is not in the document]
Comment (orgads) [reply engines/scumm/string.cpp]: Rebased and force-pushed for you.

File: engines/scumm/scumm.cpp [hunk 3]
```diff
@@ -312,7 +312,7 @@ ScummEngine::ScummEngine(OSystem *syst, const DetectorResult &dr)
 		_herculesBuf = (byte *)malloc(kHercWidth * kHercHeight);
 	}
 
-	_isRTL = (_game.version >= 4 && _game.version < 7 && _game.heversion == 0 && _language == Common::HE_ISR);
+	_isRTL = _game.id == GID_MANIAC || (_game.version >= 4 && _game.version < 7 && _game.heversion == 0 && _language == Common::HE_ISR);
```
Comment (bluegr) [inline engines/scumm/scumm.cpp:315]: Please check it again, this change is definitely wrong

Comment (orgads) [reply engines/scumm/scumm.cpp:315]: Oops, thanks

**Comment by bluegr** [comment]
Thanks! This is good to be merged now

**Review by orgads — APPROVED** [review]

File: engines/scumm/string.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (ShaharAriel) [reply engines/scumm/string.cpp]: Thanks.

**Pull request merged**